src/Development/Rattle/Server.hs

```diff
@@ -146,6 +146,10 @@ recoverableHazard :: Hazard -> Bool
 recoverableHazard WriteWriteHazard{} = False
```

Comment:
Is this now wrong?

Reply:
No, because writewrite hazards are never recoverable so there is no need to check.

Reply:
That seems like a detail of write hazards we shouldn't be assuming here? It feels wrong to put the data in the field, and then not check it. Of course, if we go with `hazardRecoverable` it just falls out nicely.

Reply:
I guess I was just trying to enforce the invariant by having it always return false, but of course this wont be an issue like you said if we make the change you suggested.